controllers/hadithSearch.controller.js

```diff
@@ -50,55 +24,84 @@ class HadithSearchController {
       });
     }
 
-    const response = await fetchWithTimeout(url);
-    const data = await response.json();
+    let response;
+    try {
+      response = await fetchWithTimeout(url);
+    } catch (error) {
+      if (
+        error.statusCode === 404 ||
+        error.status === 404 ||
+        error.message.includes('Not Found')
+      ) {
+        throw new AppError('Hadith not found', 404);
+      }
+      throw error;
+    }
```

Comment:
You try-catch wrapper are already used in `catchAsync` file in utils folder
so if `fetchWithTimeout` throw an error, it will catch in `catchAsync`
you can take a look and put any logic you want there 

also test it